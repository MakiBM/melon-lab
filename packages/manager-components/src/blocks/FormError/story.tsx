import { storiesOf } from '@storybook/react';
import React from 'react';
import FormError from './index';

storiesOf('Blocks|Form Error', module)
  .add('Default', () => {
    return <FormError>Default button</FormError>;
  });
