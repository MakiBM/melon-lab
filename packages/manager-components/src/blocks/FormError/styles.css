.form-error {
  font-size: var(--font-size-s);
  color: var(--color-loss);
}
