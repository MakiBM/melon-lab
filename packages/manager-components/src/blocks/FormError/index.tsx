import React, { StatelessComponent } from 'react';

import styles from './styles.css';

const FormError: StatelessComponent = ({ children }) => {
  return (
    <div className="form-error">
      <style jsx>{styles}</style>
      {children}
    </div>
  );
};

export default FormError;
