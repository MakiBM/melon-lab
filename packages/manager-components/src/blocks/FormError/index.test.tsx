import React from 'react';
import FormError from './index';

describe('Button', () => {
  const defaultElement = <FormError>Default error</FormError>;

  it('should render correctly', () => {
    const wrapper = shallow(defaultElement);
    expect(wrapper).toMatchSnapshot();
  });

  it('should be selectable by class "form-error"', () => {
    expect(shallow(defaultElement).is('.form-error')).toBe(true);
  });

  it('should mount in a full DOM', () => {
    expect(mount(defaultElement).find('.form-error').length).toBe(1);
  });

  it('should render to static HTML', () => {
    expect(render(defaultElement).text()).toEqual('Default error');
  });
});
