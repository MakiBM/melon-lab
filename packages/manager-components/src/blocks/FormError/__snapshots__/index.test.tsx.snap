// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render correctly 1`] = `
<div
  className="jsx-__scopedHash form-error"
>
  <JSXStyle
    css="__scoped"
    styleId="__scopedHash"
  />
  Default error
</div>
`;
